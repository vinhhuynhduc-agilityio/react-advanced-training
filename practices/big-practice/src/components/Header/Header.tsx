import React from "react";

// component
import Button from "../Button/Button";

interface HeaderProps {
  onAddUser: () => void;
};

const Header: React.FC<HeaderProps> = ({
  onAddUser
}) => {
  const handleAddTask = () => {
    console.log("Add a task clicked!");
  };

  const handleAddProject = () => {
    console.log("Add a task clicked!");
  };

  return (
    <header className=" bg-white flex flex-row py-2 px-3 items-center border-2 border-customBorder">
      <div className="flex flex-row items-center w-[16.5rem]">
        <img src="/src/assets/logo.png" alt="Logo" className="h-12 w-12 mr-2" />
        <h1 className=" text-xl font-medium text-[#475466]">
          Team Progress
        </h1>
      </div>
      <Button label="Add a user" onClick={onAddUser} />
      <Button label="Add a task" onClick={handleAddTask} />
      <Button label="Add a project" onClick={handleAddProject} />
    </header>
  );
};

export default Header;
