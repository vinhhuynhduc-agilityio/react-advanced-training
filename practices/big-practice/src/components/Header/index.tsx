import React, { memo } from 'react';

// component
import { Button } from '@/components/common';

interface HeaderProps {
  onAddUser: () => void;
  onAddProject: () => void;
  onAddTask: () => void;
  isLoading: boolean;
  isSavingUser: boolean;
  isSavingTask: boolean;
  isSavingProject: boolean;
};

export const Header: React.FC<HeaderProps> = memo(
  ({
    onAddUser,
    onAddProject,
    onAddTask,
    isLoading,
    isSavingUser,
    isSavingTask,
    isSavingProject
  }) => {
    const isDisable = isLoading ||
      isSavingUser ||
      isSavingTask ||
      isSavingProject

    return (
      <header className=" bg-white flex flex-row py-2 px-3 items-center border border-customBorder">
        <div className="flex flex-row items-center w-[16.5rem]">
          <img src="/assets/logo.png" alt="Logo" className="h-12 w-12 mr-2" />
          <h1 className=" text-xl font-medium text-[#475466]">
            Team Progress
          </h1>
        </div>
        <Button
          label="Add a user"
          onClick={onAddUser}
          disabled={isDisable}
        />
        <Button
          label="Add a task"
          onClick={onAddTask}
          disabled={isDisable}
        />
        <Button
          label="Add a project"
          onClick={onAddProject}
          disabled={isDisable}
        />
      </header>
    );
  }
);
