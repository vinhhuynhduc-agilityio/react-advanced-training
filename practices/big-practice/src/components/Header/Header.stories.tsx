import type { Meta, StoryObj } from '@storybook/react';
import { Header } from '@/components/Header';

// Metadata for story
const meta: Meta<typeof Header> = {
  title: 'Components/Header',
  component: Header,
  tags: ['autodocs'],
  parameters: {
    layout: 'fullscreen',
  },
  argTypes: {
    onAddUser: { action: 'Add User Clicked' },
    onAddProject: { action: 'Add Project Clicked' },
    onAddTask: { action: 'Add Task Clicked' },
  },
};

export default meta;
type Story = StoryObj<typeof Header>;

// Story default
export const Default: Story = {};
