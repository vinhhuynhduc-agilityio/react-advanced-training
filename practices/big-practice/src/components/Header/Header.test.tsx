import { render, screen } from '@testing-library/react';
import Header from './Header';

describe('Header Component', () => {
  const mockOnAddUser = jest.fn();
  const mockOnAddProject = jest.fn();
  const mockOnAddTask = jest.fn();

  it('matches snapshot for default state', () => {
    const { container } = render(
      <Header
        onAddUser={mockOnAddUser}
        onAddProject={mockOnAddProject}
        onAddTask={mockOnAddTask}
      />
    );;
    expect(container).toMatchSnapshot();
  });

  it('should render the logo, title, and buttons', () => {
    render(
      <Header
        onAddUser={mockOnAddUser}
        onAddProject={mockOnAddProject}
        onAddTask={mockOnAddTask}
      />
    );
    const logoElement = screen.getByAltText('Logo');
    expect(logoElement).toBeInTheDocument();
    expect(logoElement).toHaveAttribute('src', '/assets/logo.png');

    const titleElement = screen.getByText('Team Progress');
    expect(titleElement).toBeInTheDocument();

    const addTaskButton = screen.getByText('Add a task');
    expect(addTaskButton).toBeInTheDocument();

    const addProjectButton = screen.getByText('Add a project');
    expect(addProjectButton).toBeInTheDocument();
  });

});
