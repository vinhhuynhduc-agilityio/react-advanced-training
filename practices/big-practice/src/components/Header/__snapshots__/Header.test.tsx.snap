// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches snapshot for default state 1`] = `
<div>
  <header
    class=" bg-white flex flex-row py-2 px-3 items-center border border-customBorder"
  >
    <div
      class="flex flex-row items-center w-[16.5rem]"
    >
      <img
        alt="Logo"
        class="h-12 w-12 mr-2"
        src="/src/assets/logo.png"
      />
      <h1
        class=" text-xl font-medium text-[#475466]"
      >
        Team Progress
      </h1>
    </div>
    <button
      class="px-4 py-2 border-none mr-7 bg-[#F4F5F9] text-[#1CA1C1] hover:bg-slate-300"
    >
      Add a user
    </button>
    <button
      class="px-4 py-2 border-none mr-7 bg-[#F4F5F9] text-[#1CA1C1] hover:bg-slate-300"
    >
      Add a task
    </button>
    <button
      class="px-4 py-2 border-none mr-7 bg-[#F4F5F9] text-[#1CA1C1] hover:bg-slate-300"
    >
      Add a project
    </button>
  </header>
</div>
`;
